Model/EmailDestinationInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Lexik\Bundle\MailerBundle\Model;
+
+/**
+ * EmailDestinationInterface to be applied to user entities
+ *
+ */
+interface EmailDestinationInterface
```

Comment:
Maybe these could be named `public function getRecipientName();` and `public function getRecipientAddress();`?

I feel `getName()` and `getEmail()` are incredibly generic and might conflict with existing implementations.